acme/src/acme/_internal/tests/errors_test.py

```diff
@@ -51,5 +51,27 @@ def test_repr(self):
                          'sentinel.AR2})' % repr(set()) == repr(self.invalid)
 
 
+class ValidationErrorTest(unittest.TestCase):
+    """Tests for acme.errors.ValidationError"""
+
+    def setUp(self):
+        from acme.errors import ValidationError
+        from acme.messages import Error
+        failed_authzr = mock.MagicMock()
+        failed_authzr.body.identifier = 'example.com'
```

Comment:
this should either be changed to
```suggestion
        failed_authzr.body.identifier.value = 'example.com'
```
or use a (potentially mock) [identifier object](https://github.com/certbot/certbot/blob/0075104805e2e4fd9cf0e6ec69d80ab203e03655/acme/src/acme/messages.py#L111-L119)

tests passing like this yet not behaving as we expect personally motivates me a bit to avoid mock objects, but creating real objects for everything here is certainly quite a pain unfortunately

Reply:
yeah i'm not a huge fan of using mocks either. i'll see how straightforward it'd be to get an actual error object here